packages/next/src/server/app-render/app-render.tsx

```diff
@@ -3125,33 +3170,34 @@ async function prerenderToStream(
           res.statusCode === 404
         )
 
-        const serverPrerenderStreamResult = await prerenderServerWithPhases(
-          finalServerController.signal,
-          () =>
-            workUnitAsyncStorage.run(
-              finalServerPrerenderStore,
-              ComponentMod.renderToReadableStream,
-              finalServerPayload,
-              clientReferenceManifest.clientModules,
-              {
-                onError: (err: unknown) => {
-                  if (finalServerController.signal.aborted) {
-                    serverIsDynamic = true
-                    if (isPrerenderInterruptedError(err)) {
-                      return err.digest
+        const serverPrerenderStreamResult = (reactServerPrerenderResult =
+          await prerenderServerWithPhases(
+            finalServerController.signal,
+            () =>
+              workUnitAsyncStorage.run(
+                finalServerPrerenderStore,
+                ComponentMod.renderToReadableStream,
+                finalServerPayload,
+                clientReferenceManifest.clientModules,
+                {
+                  onError: (err: unknown) => {
+                    if (finalServerController.signal.aborted) {
+                      serverIsDynamic = true
+                      if (isPrerenderInterruptedError(err)) {
+                        return err.digest
+                      }
```

Comment:
why doesn't this do the `getDigestForWellKnownError` thing too?

Reply:
Here it's handled already as part of `serverComponentsErrorHandler` two lines below this one.

Reply:
Oh you mean in case rendering is interrupted already. Yeah, I think we might need to use it then as well. Need to think about a test case that triggers this branch...

Reply:
I don't think this can happen. When `finalServerController.abort()` is called, React will abort the prerender and call `onError` once with the `AbortError`, and after that it won't be called anymore.

This nested `if` (existed before) might even be wrong and the inner one might never be reached. In other places we use this instead:

```js
if (isPrerenderInterruptedError(err) || controller.signal.aborted) {
  // ...
}
```

@gnoff Do you agree with this assessment?

Reply:
in RSC for the final render there is only one controller so when we interrupt the render we sync abort the controller. The reason we have the `isPrerenderInterruptedError()` as a sibling of the aborted check in the client case is for when the interrupt error in the RSC payload is observed by the SSR runtime. In this case we don't sync abort the client controller b/c the error was in the other environment but it means we are still dynamic in the SSR context.

So I think this branch can be made to just be `getDigestForWellKnowError`. Basically in this context it's not possible for the RSC prerender to be interrupted without it also being aborted